frontend_tests/node_tests/message_list.js

```diff
@@ -182,6 +182,18 @@ var MessageList = require('js/message_list').MessageList;
     assert.equal(list.nth_most_recent_id(4), -1);
 }());
 
+(function test_change_message_id() {
+    var table;
+    var filter = {};
+
+    var list = new MessageList(table, filter);
+    list.append([{id:10, content: "good job"}, {id:20, content: "ok!"}]);
+    list.change_message_id(10, 11);
+    assert.equal(list.get(11).content, "good job");
+
+    list.change_message_id(20, 10);
+    assert.equal(list.get(10).content, "ok!");
+}());
```

Comment:
we typically use `change_message_id` with non-integer initial IDs (from local echo); might make sense to od that here for consistency.